### ISSUE TITLE
Double-clicking the title bar does not zoom the window

### ISSUE THREAD

**Issue by J-rg** [desktop/desktop#12884]
### Describe the bug
In OS X Yosemite, the action of the green button in the title bar changed from Zoom to Enter Full Screen and a setting that allows for zooming the window by double-clicking the title bar was introduced, which is enabled by default. GitHub Desktop currently does not respect this setting.

### Version & OS
Version 2.9.3 (x64), macOS 11.5.2 (20G95)

### Steps to reproduce the behavior
1. Double-click the title bar.

### Expected behavior
Same behavior as clicking on the green button while holding the alt key or selecting Window → Zoom from the main menu, i.e. either maximizing the window or restoring its previous dimensions.

### Actual behavior
Nothing happens.

### Screenshots
![Double-click a window’s title bar to zoom](https://user-images.githubusercontent.com/4042863/131677651-4b0261cd-2497-4354-a814-e9fec44069db.png)

**Comment by steveward** [comment]
Thanks for the report @J-rg. We had a previous case of this in https://github.com/desktop/desktop/issues/10594. I'm unable to reproduce this behavior (running macOS 11.5.2 as well).

There are some [debugging steps in the comment](https://github.com/desktop/desktop/issues/10594#issuecomment-698908604) -- it'd be helpful if you could check the Developer Tools to see what that value returns on your machine.

**Comment by J-rg** [comment]
At first it returned an empty string, but after I toggled the option on and off, it now returns “Maximize” and GitHub Desktop behaves as expected.
I then created a fresh user account and noticed the same issue, no zoom and empty return, there is probably no AppleActionOnDoubleClick key in NSUserDefaults if the option was never changed.
So I guess [this line](https://github.com/desktop/desktop/blob/586ae2f6e40c1e2e5e67a1b7404ea68df99ac0b0/app/src/ui/window/title-bar.tsx#L62) should also check for empty strings?
You might be able to reproduce the issue by typing `defaults delete NSGlobalDomain AppleActionOnDoubleClick` into Terminal.
